### PR TITLE
Re-add directory mode 

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: py2many/cli.py [hunk 1]
```diff
@@ -131,11 +140,40 @@ def main():
...
             outdir = pathlib.Path(args.outdir)
-        print(f"Writing to: {outdir}")
-        output_path = outdir / (source.stem + settings.ext)
-        print(f"{filename}...{output_path}")
-        with open(output_path, "w") as f:
-            source_data = open(source).read()
-            f.write(transpile(source_data, settings.transpiler))
-        os.system(f"{settings.formatter} {output_path}")
-        print()
+        if os.path.isfile(filename):
```
Comment (adsharma) [inline py2many/cli.py:143]:
```
source.is_file()
```

Some of the code below reads a lot nicer with pathlib:

https://realpython.com/python-pathlib/

File: py2many/cli.py [hunk 2]
```diff
@@ -131,11 +140,40 @@ def main():
...
+            print(f"Writing to: {outdir}")
+            _process_once(settings, source, outdir)
+        else:
+            basename = os.path.basename(filename)
+            if args.outdir is None:
+                outdir = source.parent / f"{basename}-py2many"
```
Comment (adsharma) [inline py2many/cli.py:149]:
```
 outdir = source.parent / f"{source.name}-py2many"
 ```

File: py2many/cli.py [hunk 3]
```diff
@@ -131,11 +140,40 @@ def main():
...
+            print(f"Transpiling whole directiory to {outdir}:")
+            successful = failures = format_errors = 0
+            for root, subdir, filenames in os.walk(filename):
```
Comment (adsharma) [inline py2many/cli.py:152]: Could be written is using source.glob("*.py")

File: py2many/cli.py [hunk 4]
```diff
@@ -90,6 +90,15 @@ def go_settings(args):
     return LanguageSettings(GoTranspiler(), ".go", "gofmt -w")
 
 
+def _process_once(settings, filename, outdir):
+    output_path = outdir / (filename.stem + settings.ext)
+    print(f"{filename}...{output_path}")
+    with open(output_path, "w") as f:
+        source_data = open(filename).read()
```
Comment (adsharma) [inline py2many/cli.py:97]: File handle leak here (leak present in main branch as well). Using a context manager should fix.

**Review by adsharma — CHANGES_REQUESTED** [review]
Would love to have the directory mode.

File: py2many/cli.py (hunk 1 above)
Comment (jayvdb) [reply py2many/cli.py]: done

**Pull request merged**